dissect/cstruct/types/pointer.py

```diff
@@ -74,7 +78,7 @@ def _write(cls, stream: BinaryIO, data: int) -> int:
         return cls.cs.pointer._write(stream, data)
 
     def dereference(self) -> Any:
-        if self == 0:
+        if self == 0 or self._stream is None:
```

Comment:
Is it  a valid state to have `self == 0 and self._stream is not None`, or 
`self != 0 and self._stream is None`?

If not, perhaps we should assert this invariant earlier during construction of the pointer?

Reply:
Yes, those are valid states. 

`self == 0 and self._stream is not None`: we have a source stream, but the read/parsed pointer value is 0.
`self != 0 and self._stream is None`: We initialize an "empty" pointer instance with a value, but we don't have a source stream. This is a valid state for writing.